### PR TITLE
Use babel for esm for core

### PR DESCRIPTION
**Pull request by bertya**
WIP of tree shaking.

Currently, only affect core package. 

The structure is not final, some files are redundant. Just demonstrate how it works.

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -3,9 +3,10 @@
   "version": "2.0.4",
   "description": "Priceline Design System",
   "main": "dist/index.cjs.js",
-  "module": "dist/index.esm.js",
+  "module": "dist/index.js",
+  "sideEffects": false,
   "scripts": {
-    "prepare": "rollup --config",
+    "prepare": "babel src/ --out-dir dist --ignore spec.js,story.js; rollup --config",
```
Comment (BeniCheni) [inline packages/core/package.json]: Wonder why you prefer `;` over `&&`. Read a bit on topic in a somewhat [older stackexchange post](https://unix.stackexchange.com/questions/37069/what-is-the-difference-between-and-when-chaining-commands). Would `&&` work also? If for `;` what returned value from first `babel` would the second `rollup` command depend on?

Comment (bertya) [reply packages/core/package.json]: I actually don't think it really matters here. I'm open to either.

Comment (BeniCheni) [inline packages/core/package.json]: @bertya, think for consistency to existing chained commands in npm-run scripts, and the fact that there'd be a subtil diff in the `;` and `&&` in the unix implication, I'd prefer to align to use `&&. Proposed the edit.

```suggestion
    "prepare": "babel src/ --out-dir dist --ignore spec.js,story.js && rollup --config",
```

**Comment by BeniCheni** [comment]
Confirmed w. @bertya that this PR is outdated to be closed.

**Pull request closed without merging**